File: db/csharp/SalesManagement.Infrastructure/Migrations/Migration_20250106_001_InitialSetup.cs
using FluentMigrator;

namespace SalesManagement.Infrastructure.Migrations
{
    /// <summary>
    /// 初期セットアップマイグレーション
    /// </summary>
    [Migration(20250106001)]
    public class Migration20250106001InitialSetup : Migration
    {
        public override void Up()
        {
            // PostgreSQL用のタイムゾーン設定
            IfDatabase("Postgres")
                .Execute.Sql("SET timezone = 'Asia/Tokyo'");

            // MySQL用のタイムゾーン設定
            IfDatabase("MySQL")
                .Execute.Sql("SET time_zone = '+09:00'");
        }

        public override void Down()
        {
            // ロールバック処理は不要
        }
    }
}

File: db/fsharp/SalesManagement.Infrastructure/Migrations/Migration_20250106_001_InitialSetup.fs
namespace SalesManagement.Infrastructure.Migrations

open FluentMigrator

/// <summary>
/// 初期セットアップマイグレーション
/// </summary>
[<Migration(20250106001L)>]
type Migration_20250106_001_InitialSetup() =
    inherit Migration()

    override this.Up() =
        // PostgreSQL用のタイムゾーン設定
        this.Execute.Sql("SET timezone = 'Asia/Tokyo'")

    override this.Down() =
        // ロールバック処理は不要
        ()
